### ISSUE TITLE
Migration from non-FIPS to FIPS BC jar

### ISSUE THREAD

**Issue by SushmaReddyLoka** [bcgit/bc-java#1534]
We have an application which has data already encrypted using non-FIPS Bouncy castle jar which is “bcprov-jdk15on-1.67.jar” with algorithm as “AES/CBC/ PKCS7Padding”, now we want to make this application FIPS complaint , we would be continuing to use the same algorithm without loosing the data which is already encrypted using non-FIPS BC . 

I tested this locally and it seems to be working, looking for any official article/documentation from Bouncy castle confirming no data loss during this migration.

**Comment by ecki** [comment]
non official answer: There are condition when it works (if you use the same algorithm and mode it should work, however some modes might not be supported by Fips or some might be more strict about maximum size or key reuse. I think this can be derived from the FIPS manual in most cases but there cant be a central list of issues as its a complex topic.

**Comment by SushmaReddyLoka** [comment]
Yes we plan to use the algorithm and mode that was used while encrypting the data at the time of decryption as well.
 As I mentioned we are using "AES/CBC/PKCS7Padding" with key sizes as 128, 256 and I see this algorithms are FIPS complaint. So decryption should work without any issues in this case.

**Comment by SushmaReddyLoka** [comment]
Yes we plan to use the algorithm and mode that was used while encrypting
the data at the time of decryption as well.
As I mentioned we are using "AES/CBC/PKCS7Padding" with key sizes as 128,
256 and I see this algorithms are FIPS complaint. So decryption should work
without any issues in this case.

**Comment by ecki** [comment]
It should but you never know, for example (i.e. without looking at the specific cipher, something I guess Keyfactor support contract would cover) the padding might be stricter in failing for truncated or excessive bytes. But your testing should reveal that.

**Comment by dghgit** [comment]
Yes, the algorithm should be fine, both BC and BCFIPS pass the same test vectors for this, which I guess explains why this has gone silent. Closing as inactive.
